omxplayer.cpp

```diff
@@ -1736,8 +1748,19 @@ int main(int argc, char *argv[])
       {
         if (m_av_clock->OMXIsPaused())
         {
-          CLog::Log(LOGDEBUG, "Resume %.2f,%.2f (%d,%d,%d,%d) EOF:%d PKT:%p\n", audio_fifo, video_fifo, audio_fifo_low, video_fifo_low, audio_fifo_high, video_fifo_high, m_omx_reader.IsEof(), m_omx_pkt);
-          m_av_clock->OMXResume();
+          if (!m_start_paused)
+          {
+            CLog::Log(LOGDEBUG, "Resume %.2f,%.2f (%d,%d,%d,%d) EOF:%d PKT:%p\n", audio_fifo, video_fifo, audio_fifo_low, video_fifo_low, audio_fifo_high, video_fifo_high, m_omx_reader.IsEof(), m_omx_pkt);
+            m_av_clock->OMXResume();
+          }
+          else
+          {
+            CLog::Log(LOGDEBUG, "start-paused(%d)\n", m_start_paused);
+            m_av_clock->OMXResume();
+            m_Pause=true;
+            m_start_paused=false;
+            m_av_clock->OMXPause();
```

Comment:
From my own experiments, I've found that only `m_Pause=true;` is needed, and the player will see that variable has changed and pause once it loops around.

I found that including `m_av_clock->OMXResume();` and `m_av_clock->OMXPause();` may cause unwanted effects. In my case, it seemed that the audio stream occasionally stopped, while the video stream continued when using `--start-paused`. 

Using my suggested change below seemed to have stopped this behaviour for me
 
```suggestion
            m_Pause=true;
            m_start_paused=false;
```

Reply:
@DazB Could you update your prepare-native-raspbian.sh script? Having trouble here trying to compile... thank you.

Reply:
Hi @zasnicoff. That's also an issue with omxplayer, and not to do with any changes I've made. I too had those issues when first compiling. I probably should update my own fork of prepare-native-raspbian.sh to remedy this, but since it only happens when first compiling I didn't get round to it. Appreciate the heads up.